platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -415,22 +418,22 @@ public static MetricRegistry getMetricRegistry() {
   }
 
   public static MeterRegistry getMeterRegistry() {
-    return METER_REGISTRY;
+    return meterRegistry;
   }
 
   public static synchronized void stop() {
     stopConsoleMetricsReporter();
     METRIC_REGISTRY.getNames().forEach(METRIC_REGISTRY::remove);
 
-    DEFAULT_TAGS.clear();
     /* For each meter registry in this composite, it will call the close function */
-    METER_REGISTRY.getRegistries().forEach(MeterRegistry::close);
-    METER_REGISTRY.forEachMeter(METER_REGISTRY::remove);
-    METER_REGISTRY.getRegistries().forEach(MeterRegistry::clear);
-    Set<MeterRegistry> registries = new HashSet<>(METER_REGISTRY.getRegistries());
-    registries.forEach(METER_REGISTRY::remove);
+    meterRegistry.getRegistries().forEach(MeterRegistry::close);
+    meterRegistry.forEachMeter(meterRegistry::remove);
```

Comment:
nit - a lot of this code - 430-434 I believe - is redundant now that we're just replacing the whole registry. We don't need to go through and clear stuff out any more.